### PR TITLE
Cli parse

### PR DESCRIPTION
**Pull request by pirat89**
Improve the cmdline parsing

Major changes:

* the filename parameter is not positional anymore

  Originally the input filename had to be given as the first parameter
  on the cmdline. The position is not required now. So the filename
  can be specified e.g. on the last position.

* the processed cmdline data is avilable via the `config` global variable

  See the `Configuration` struct

Additional minor changes:

* exit properly when the file cannot be opened (instead of sefgault :eyes: )
* the cmdline is processed inside the `process_cmdline` function
* ignore the *.test-playground* directory

  We decided to create a directory to have a spece where we could put own
  testing data, notes, ... and do not want to make it part of the git repo.
* comment the code better

### CODE REVIEW

File: wc.c [hunk 1]
```diff
@@ -23,34 +26,98 @@ void print_help() {
...
 
-int main(int argc, char *argv[]) {
+struct Configuration {
+    bool char_counter;    // character counter
+    bool line_counter;    // line counter
+    bool word_counter;    // word counter
+
+    char* separator;      // define own separator for counting
+    char* filename;       // the file that should be parsed
+} config = {false, false, false, NULL};
```
Comment (viktormalik) [inline wc.c]:
```suggestion
} config = {false, false, false, NULL, NULL};
```

**Review by viktormalik — CHANGES_REQUESTED** [review]

File: wc.c (hunk 1 above)
Comment (pirat89) [reply wc.c]: Thanks for the catch! Fixed.

File: wc.c [hunk 2]
```diff
@@ -23,34 +26,98 @@ void print_help() {
...
+            continue;
+        } else if (strcmp("-s", argv[i]) == 0) {
+            if (i+1 >= argc) {
+              fprintf(stderr, "The '-s' option requires argument.\n");
+              return 1;
+            }
+            i++;
+            config.separator = argv[i];
+            count_defined++;
+            continue;
```
Comment (viktormalik) [inline wc.c]: `continue` statements are redundant

**Review by viktormalik — CHANGES_REQUESTED** [review]

File: wc.c (hunk 2 above)
Comment (pirat89) [reply wc.c]: Bad me! Thanks.. fixed

File: wc.c [hunk 3]
```diff
@@ -23,34 +26,89 @@ void print_help() {
...
         return 1;
     }
-    FILE *f = fopen(argv[1], "r");
 
-    int c_opt = strcmp("-c", argv[2]) == 0;
-    int l_opt = strcmp("-l", argv[2]) == 0;
-    int w_opt = strcmp("-w", argv[2]) == 0;
-    int s_opt = strcmp("-s", argv[2]) == 0;
+    int count_defined = 0;
+    for (int i=1; i < argc; i++) {
```
Comment (viktormalik) [inline wc.c]:
```suggestion
    for (int i = 1; i < argc; i++) {
```

File: wc.c [hunk 4]
```diff
@@ -23,34 +26,89 @@ void print_help() {
 }
 
 
-int main(int argc, char *argv[]) {
+struct Configuration {
+    bool char_counter;    // character counter
+    bool line_counter;    // line counter
+    bool word_counter;    // word counter
+
+    char* separator;      // define own separator for counting
```
Comment (viktormalik) [inline wc.c]: separator could be a `char`

**Review by viktormalik — CHANGES_REQUESTED** [review]

**Review by viktormalik — APPROVED** [review]

**Pull request merged**